### PR TITLE
dotnet-scaffold-aspire with some commands to add packages.

### PR DESCRIPTION
**Pull request by deepchoudhery**
dotnet-scaffold-aspire now has 3 commands : `caching`, `database`, and `storage`. Some info about these below:
- currently the commands install the appropriate packages using `DotnetCommands.AddPackage`
- all have `--type` and `--host-project` options. Help with which packages to install
- have an `--web-project` or `--api-project` for the other associated package to install packages to.
- later will have Program.cs changes and additional files being added.
- `Microsoft.DotNet.Tools.Scaffold.Aspire.Helpers.PackageConstants` contains all needed package names and dictionaries to help with scenarios.

Other changes:
- added `InteractivePickerType.CustomPicker` to allow `Parameter` to force the user to pick from a set of custom values.
- removed any `IEnvironmentVariableProvider`, not needed currently and was throwing commandline execution off.

### CODE REVIEW

File: tools/dotnet-scaffold-aspire/Commands/CachingCommand.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        public CachingCommand(IFileSystem fileSystem, ILogger logger, IAppSettings appSettings)
+        {
+            _appSettings = appSettings;
+            _fileSystem = fileSystem;
+            _logger = logger;
+        }
+
+        public override int Execute([NotNull] CommandContext context, [NotNull] CachingCommandSettings settings)
+        {
+            Debugger.Launch();
```
Comment (tlmii) [inline tools/dotnet-scaffold-aspire/Commands/CachingCommand.cs:30]: Helpful for my debugging purposes but I assume you don't mean to leave this in here :)

File: tools/dotnet-scaffold-aspire/Commands/GetCommands.cs [hunk 2]
```diff
@@ -33,7 +49,17 @@ public override int Execute(CommandContext context)
 
     public static class GetCmdsHelper
     {
-        internal static Parameter ProjectParameter = new() { Name = "--project", DisplayName = "Project Name", Description = "Project of choice for the scaffolding", Required = true, Type = BaseTypes.String, PickerType = InteractivePickerType.ProjectPicker };
-        internal static Parameter[] RedisParameters = [ProjectParameter];
+        internal static Parameter HostProjectParameter = new() { Name = "--host-project", DisplayName = "Aspire Host project file", Description = "Aspire host app for the scaffolding", Required = true, Type = BaseTypes.String, PickerType = InteractivePickerType.ProjectPicker };
```
Comment (tlmii) [inline tools/dotnet-scaffold-aspire/Commands/GetCommands.cs:52]: `Host` -> `AppHost`

File: tools/dotnet-scaffold-aspire/Commands/GetCommands.cs [hunk 3]
```diff
@@ -33,7 +49,17 @@ public override int Execute(CommandContext context)
...
+        internal static Parameter ApiProjectParameter = new() { Name = "--api-project", DisplayName = "API project file", Description = "API project associated with the Aspire Starter App", Required = true, Type = BaseTypes.String, PickerType = InteractivePickerType.ProjectPicker };
```
Comment (tlmii) [inline tools/dotnet-scaffold-aspire/Commands/GetCommands.cs:53]: I have questions/thoughts about these other parameter names but don't want to block the PR on them, we can discuss and change later.

**Review by tlmii — APPROVED** [review]
Approving, with just a few things to change/tweak real quick and a couple questions. Getting there and really starting to look good!

File: scripts/install-scaffold.cmd [hunk 4]
```diff
@@ -4,7 +4,7 @@ set SRC_DIR=%cd%
 set NUPKG=artifacts/packages/Debug/Shipping/
 call taskkill /f /im dotnet.exe
 call rd /Q /S artifacts
-call build
+call dotnet pack tools\dotnet-scaffold\dotnet-scaffold.csproj -c Debug
```
[the remark this answers is not in the document]
Comment (deepchoudhery) [reply scripts/install-scaffold.cmd:7]: `build` was building the entire repo. Only need the packages from the dotnet-scaffold.csproj. and `dotnet build` would not create the needed packages for install.

File: tools/dotnet-scaffold-aspire/Commands/CachingCommand.cs (hunk 1 above)
Comment (deepchoudhery) [reply tools/dotnet-scaffold-aspire/Commands/CachingCommand.cs:30]: oops will remove tyty

File: tools/dotnet-scaffold-aspire/Commands/GetCommands.cs (hunk 2 above)
Comment (deepchoudhery) [reply tools/dotnet-scaffold-aspire/Commands/GetCommands.cs:52]: yes will change

File: tools/dotnet-scaffold-aspire/Commands/GetCommands.cs (hunk 3 above)
Comment (deepchoudhery) [reply tools/dotnet-scaffold-aspire/Commands/GetCommands.cs:53]: sounds good tyty

File: src/Shared/Microsoft.DotNet.Scaffolding.Helpers/General/DotnetCommands.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                arguments.Add(projectFile);
+            }
+
+            arguments.AddRange(["package", packageName]);
+            if (!string.IsNullOrEmpty(packageVersion))
+            {
+                arguments.AddRange(["-v", packageVersion]);
+            }
+
+            arguments.Add("--prerelease");
```
[the remark this answers is not in the document]
Comment (deepchoudhery) [reply src/Shared/Microsoft.DotNet.Scaffolding.Helpers/General/DotnetCommands.cs:27]: yep i'm thinking how to manage this fact. I was thinking likely a tfm check, `prerelease` for net9.0 (which is not GA yet) and net8.0 looking for GA packages. Assuming most Aspire packages will GA soon (not sure if they will have different naming for net9.0 and net8.0 packages)

**Pull request merged**